client/my-sites/marketplace/components/review-item/create-review-item.tsx

```diff
@@ -0,0 +1,167 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { Button, ConfettiAnimation } from '@automattic/components';
+import Card from '@automattic/components/src/card';
+import { CheckboxControl, TextareaControl } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import moment from 'moment';
+import { useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+import Gravatar from 'calypso/components/gravatar';
+import ReviewsRatingsStars from 'calypso/components/reviews-rating-stars/reviews-ratings-stars';
+import {
+	ProductDefinitionProps,
+	useCreateMarketplaceReviewMutation,
+	useMarketplaceReviewsQuery,
+} from 'calypso/data/marketplace/use-marketplace-reviews';
+import { getCurrentUser } from 'calypso/state/current-user/selectors';
+import './style.scss';
+
+type MarketplaceCreateReviewItemProps = {
+	forceShowThankYou?: number;
+} & ProductDefinitionProps;
+
+export function MarketplaceCreateReviewItem( props: MarketplaceCreateReviewItemProps ) {
+	const { productType, slug, forceShowThankYou = 0 } = props;
+	const translate = useTranslate();
+	const currentUser = useSelector( getCurrentUser );
+	const [ content, setContent ] = useState< string >( '' );
+	const [ rating, setRating ] = useState< number >( 0 );
+	const [ errorMessage, setErrorMessage ] = useState( '' );
+	const [ showThankYouSection, setShowThankYouSection ] = useState( false );
+	const [ showContentArea, setShowContentArea ] = useState( false );
+	const [ triggerConfetti, setTriggerConfetti ] = useState( false );
+
+	const { data: userReviews, isFetching: isFetchingUserReviews } = useMarketplaceReviewsQuery( {
+		productType,
+		slug,
+		perPage: 1,
+		author: currentUser?.ID ?? undefined,
+		status: 'all',
+	} );
+
+	const resetFields = () => {
+		setShowContentArea( false );
+		setErrorMessage( '' );
+		setContent( '' );
+		setRating( 0 );
+	};
+
+	const createReviewMutation = useCreateMarketplaceReviewMutation( { productType, slug } );
+	const createReview = () => {
+		createReviewMutation.mutate(
+			{ productType, slug, content, rating },
+			{
+				onError: ( error ) => {
+					setErrorMessage( error.message );
+				},
+				onSuccess: () => {
+					resetFields();
+					setShowThankYouSection( true );
+					setTriggerConfetti( true );
+				},
+			}
+		);
+	};
+
+	const onSelectRating = ( value: number ) => {
+		setRating( value );
+		setShowContentArea( true );
+	};
+
+	// Retrigger confetti after an update
+	useEffect( () => {
+		setTriggerConfetti( false );
+		setTimeout( () => {
+			setTriggerConfetti( true );
+		}, 0 );
+	}, [ forceShowThankYou ] );
```

Comment:
An alternative solution to avoid having this `useEffect` and the `triggerConfetti` variable here https://github.com/Automattic/wp-calypso/pull/86255/files#diff-ab726e0701b8ce2f480b12b81c8f6e785c02af9ba13e699bbc39272047f7d28aR32 could be to use a key when creating the Confetti component ( [inspired by this React doc](https://legacy.reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) ). 
Something like
```jsx
<ConfettiAnimation key={ forceShowThankYou } delay={ 1000 } />
```
taking advantage of the `forceShowThankYou` variable.

This is just a piece of feedback but I don't think it changes that much the implementation, just commenting for describing an alternative. 🙁

Reply:
This is great, I love it. Thanks for the suggestion, added it here https://github.com/Automattic/wp-calypso/pull/86255/commits/5d4e2b8fa01a161bc8133e3cd84686f8b8645a24